scripts/get-all-doc-paths-of-type

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env bash
+
+# Get all adoc paths of a certain doctype in a directory
+# (default: src-documents)
+#
+
+
+log() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $*"
+} >&2
+
+err() {
+  printf "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: \e[1;31mError:\e[m %b\n" "$*"
+} >&2
+
+ecdo() {
+  log "Running: $*"
+  "$@"
+}
+
+DOCTYPE=""
+FAILED_ADOCS=()
+SUCCESS_ADOCS=()
+ADOC_REFERENCES=()
+
+print-summary() {
+  local readable_docs=$(( ${#ADOC_REFERENCES[@]} - ${#FAILED_ADOCS[@]} ))
+  log "There are ${#ADOC_REFERENCES[@]} documents in total."
+  log "Successfully picked ${#SUCCESS_ADOCS[@]} out of ${readable_docs} readable documents that are of :doctype: ${DOCTYPE}."
+  if [[ "${#FAILED_ADOCS[@]}" -gt 0 ]]
+  then
+    err "Failed to process ${#FAILED_ADOCS[@]} documents:"
+    for failure in "${FAILED_ADOCS[@]}"
+    do
+      err "  ${failure}"
+    done
+  fi
+}
+
+trap print-summary EXIT
+
+METANORMA_YAML_PATHS=()
+
+get-all-metanorma-yaml-paths() {
+  local src_dir="${1:?Missing source directory}"
+
+  while read -r yml
+  do
+    METANORMA_YAML_PATHS+=("$yml")
+  done < <(find "$src_dir" -type f -name 'metanorma.y*ml')
+
+  log "Found ${#METANORMA_YAML_PATHS[@]} metanorma YAML files"
+}
+
+get-all-sources-from-metanorma-yaml() {
+  local yml="${1:?Missing metanorma YAML file}"
+  local path_prefix="${yml%/*}"
+
+  # log "pathPrefix is $pathPrefix"
+  # Only proceed if 'files' is not null (should be an array)
+  <"$yml" yq -r ".metanorma.source.files | select(null != .) | map(\"${path_prefix}/\" + .)"
+
+}
+
+get-all-sources-from-metanorma-yamls() {
+  for yml in "${METANORMA_YAML_PATHS[@]}"
+  do
+    get-all-sources-from-metanorma-yaml "$yml"
+  done
+}
+
+# Only search within the first n lines of the document
+DOC_SEARCH_FIRST_N_LINES="${DOC_SEARCH_FIRST_N_LINES:-100}"
+
+COMMON_DOCTYPES=(
+  standard
+  directive
+  guide
+  specification
+  report
+  amendment
+  technical-corrigendum
+  administrative
+  international-standard
+  internet-draft
+  advisory
+)
+
+# See: https://github.com/CalConnect/VCARD/blob/4d17cb269cbe48cff9603d293e1374bc81fba7f1/presentations/20180926-calconnect-mn-authoring/index.adoc?plain=1#L220
+
+#   :doctype: can be standard, directive, guide, specification, report,
+#   amendment, technical-corrigendum.
+#
+# But these are also seen in the wild:
+#   administrative | international-standard | internet-draft | advisory
```

Comment:
It's because this script won't work by a simple grep. There are ISO documents and IETF documents that should not be published on our site. You need to get the files that are flavor "cc".

Reply:
By "flavor" I assume you mean `:mn-document-class:` within the adocs, correct?

Reply:
The "flavor" is official terminology. Yes that's the document attribute for setting flavor.